pages/signup.vue

```diff
@@ -0,0 +1,245 @@
+          <v-sheet
+            class="section d-flex align-center"
+            style="padding-bottom: 200px"
+          >
+            <!-- Right side for login component -->
+            <v-col
+              cols="12"
+              xs="12"
+              sm="10"
+              md="10"
+              lg="4"
+              xl="3"
+              xxl="3"
+              class="d-flex flex-column justify-start align-start"
+            >
+                <h1 class="login-text mx-auto">Sign Up</h1>
+                <div class="text-input-wrapper">
+                    <input class="duolingo-text-input" v-model="email" data-test="email-input" type="text" placeholder="Email or username">
+                    <button @click="email = ''" class="cancel-email-input-button" tabindex="-1" type="button">
+                        <img src="../assets/cancel-circle.svg">
+                    </button>
+                </div>
+                <div class="text-input-wrapper">
+                    <input class="duolingo-text-input" v-model="password" data-test="email-input" type="password" placeholder="Password">
+                </div>
+                <small class="text-red" v-if="errorOccured">Error Occured: {{ errorType }}</small>
+                <LAButton text="SIGN UP" class="w-100" margin="0px" @click="signupsubmit"/>
+                <div class="alt-login-divider">
+                    <div class="horizontal-bar"></div>
+                    <div class="upper-case-text-format or-styling">or</div>
+                    <div class="horizontal-bar"></div>
+                </div>
+                <v-row class="w-100" no-gutters>
+                    <v-col class="mr-2">
+                        <LAButton text="FACEBOOK" class="w-100" margin="0px">
+                            <img class="ml-5" src="../assets/facebook-icon.svg">
+                        </LAButton>
+                    </v-col>
+                    <v-col class="ml-2">
+                        <LAButton text="GOOGLE" class="w-100" margin="0px">
+                            <img class="ml-5" src="../assets/google-icon.svg">
+                        </LAButton>
+                    </v-col>
+                </v-row>
+            </v-col>
+          </v-sheet>
+        </v-col>
+      </v-row>
+    </v-container>
+  </template>
+  
+  <script setup>
+  import { ref } from 'vue'
+  import LAButton from '@/components/LAButton.vue'
+  import useAuth from '~/compostables/useAuth'
+  import { useRouter } from 'vue-router';
+  const { signUp } = useAuth()
+  const router = useRouter()
+  const email = ref('')
+  const password = ref('')
+  const errorOccured = ref(false)
+  const errorType = ref('')
+
+  const signupsubmit = async () => {
+    let response = await signUp(email.value, password.value) //.then(response =>{})
+    if(!response.success){
+        errorOccured.value = true
+        errorType.value = response.error
+    } else {
+        router.push('/welcome')
```

Comment:
TODO: We'll want to fetch the actual user account and route to either welcome or the topic page depending on if they have already started learning. (Welcome is just going to be that initial setup if you haven't chosen anything yet).